node/chainSimulator/components/nodeFacade.go

```diff
@@ -0,0 +1,199 @@
+package components
+
+import (
+	"errors"
+	"fmt"
+	"net"
+
+	"github.com/multiversx/mx-chain-core-go/core"
+	"github.com/multiversx/mx-chain-go/api/gin"
+	"github.com/multiversx/mx-chain-go/common"
+	"github.com/multiversx/mx-chain-go/common/forking"
+	"github.com/multiversx/mx-chain-go/config"
+	"github.com/multiversx/mx-chain-go/facade"
+	apiComp "github.com/multiversx/mx-chain-go/factory/api"
+	nodePack "github.com/multiversx/mx-chain-go/node"
+	"github.com/multiversx/mx-chain-go/node/metrics"
+	"github.com/multiversx/mx-chain-go/process/mock"
+)
+
+func (node *testOnlyProcessingNode) createFacade(configs config.Configs, enableHTTPServer bool) error {
```

Comment:
let's remove the enableHTTPServer and add more flexibility:
* define an interface called APIConfigurator that contains the following methods:
```
type APIConfigurator interface {
    Interface() string
    GetPort(shardID uint32) int // if positive, then the port will be set to that value, if the value returned is <=0, then the call to getFreePort() will be invoked
}
```
* define 3 implementations for this interface:

**noAPIConfigurator** that will return `facade.DefaultRestPortOff` when Interface method is called
**fixedPortApiConfigurator** that will receive on the constructor the interface string and a map of ports for each shard. GetPort will return the value set in the provided map
**freePortApiConfigurator** that will receive on the constructor the interface string. GetPort will return 0 (so `getFreePort()` will be invoked)

Reply:
done